### PR TITLE
[zh] Sync web page for nodelocaldns

### PR DESCRIPTION
**Pull request by chenxuc**
Sync web page for part of the items in umbrella issue:
[zh] Umbrella issue: pages out of sync in tasks section #26178

content/zh/docs/tasks/administer-cluster/nodelocaldns.md

### CODE REVIEW

File: content/zh/docs/tasks/administer-cluster/nodelocaldns.md [hunk 1]
```diff
@@ -103,57 +104,100 @@ This is the path followed by DNS Queries after NodeLocal DNSCache is enabled:
...
-
 <!--
-This feature can be enabled using the command:
+{{< note >}} The local listen IP address for NodeLocal DNSCache can be any address that can be guaranteed to not collide with any existing IP in your cluster. It's recommended to use an address with a local scope, per example, from the link-local range 169.254.0.0/16 for IPv4 or from the Unique Local Address range in IPv6 fd00::/8.
+{{< /note >}}
 -->
-可以使用以下命令启用此功能：
-
-`KUBE_ENABLE_NODELOCAL_DNS=true go run hack/e2e.go -v --up`
+{{< note >}} NodeLocal DNSCache 的本地侦听 IP 地址可以是任何地址，只要该地址不和你的集群里现有的 IP 地址发生冲突。
```
Comment (tengqm) [inline content/zh/docs/tasks/administer-cluster/nodelocaldns.md:111]:
```suggestion
{{< note >}}
NodeLocal DNSCache 的本地侦听 IP 地址可以是任何地址，只要该地址不和你的集群里现有的 IP 地址发生冲突。
```

File: content/zh/docs/tasks/administer-cluster/nodelocaldns.md [hunk 2]
```diff
@@ -103,57 +104,100 @@ This is the path followed by DNS Queries after NodeLocal DNSCache is enabled:
...
+推荐使用本地范围内的地址，例如，IPv4 链路本地区段 169.254.0.0/16 内的地址，或者 IPv6 唯一本地地址区段 fd00::/8 内的地址。
```
Comment (tengqm) [inline content/zh/docs/tasks/administer-cluster/nodelocaldns.md:112]:
```suggestion
推荐使用本地范围内的地址，例如，IPv4 链路本地区段 169.254.0.0/16 内的地址，
或者 IPv6 唯一本地地址区段 fd00::/8 内的地址。
```

File: content/zh/docs/tasks/administer-cluster/nodelocaldns.md [hunk 3]
```diff
@@ -103,57 +104,100 @@ This is the path followed by DNS Queries after NodeLocal DNSCache is enabled:
...
-在所有其他环境上，以下步骤将设置 NodeLocal DNSCache ：
+可以使用以下步骤启动此功能：
 
 <!--
-* A yaml similar to [this](https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/dns/nodelocaldns/nodelocaldns.yaml) can be applied using `kubectl create -f` command.
+* Prepare a manifest similar to the sample [`nodelocaldns.yaml`](https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/dns/nodelocaldns/nodelocaldns.yaml) and save it as `nodelocaldns.yaml.`
 -->
-* 可以使用 `kubectl create -f` 命令应用类似于[这个](https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/dns/nodelocaldns/nodelocaldns.yaml)的 Yaml 。
-
+* 根据样本 [`nodelocaldns.yaml`](https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/dns/nodelocaldns/nodelocaldns.yaml) 准备一个清单，把它保存为 `nodelocaldns.yaml`。
```
Comment (tengqm) [inline content/zh/docs/tasks/administer-cluster/nodelocaldns.md:123]:
```suggestion
* 根据示例 [`nodelocaldns.yaml`](https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/dns/nodelocaldns/nodelocaldns.yaml)
  准备一个清单，把它保存为 `nodelocaldns.yaml`。
```

File: content/zh/docs/tasks/administer-cluster/nodelocaldns.md [hunk 4]
```diff
@@ -103,57 +104,100 @@ This is the path followed by DNS Queries after NodeLocal DNSCache is enabled:
...
 <!--
-* --cluster-dns flag to kubelet needs to be modified to use the LOCAL_DNS IP that NodeLocal DNSCache is listening on (169.254.20.10 by default)
+* If using IPv6, the CoreDNS configuration file need to enclose all the IPv6 addresses into square brackets if used in IP:Port format. 
+If you are using the sample manifest from the previous point, this will require to modify [the configuration line L70](https://github.com/kubernetes/kubernetes/blob/b2ecd1b3a3192fbbe2b9e348e095326f51dc43dd/cluster/addons/dns/nodelocaldns/nodelocaldns.yaml#L70) like this `health [__PILLAR__LOCAL__DNS__]:8080`
 -->
-* 需要修改 kubelet 的 --cluster-dns 标志以使用 NodeLocal DNSCache 正在侦听的 LOCAL_DNS IP（默认为 169.254.20.10 ）
-
+* 如果使用 IPv6，在使用 IP:Port 格式的时候需要把 CoreDNS 配置文件里的所有 IPv6 地址用方括号包起来。
+如果你使用上述的样本清单，需要把 [配置行 L70](https://github.com/kubernetes/kubernetes/blob/b2ecd1b3a3192fbbe2b9e348e095326f51dc43dd/cluster/addons/dns/nodelocaldns/nodelocaldns.yaml#L70) 修改为 `health [__PILLAR__LOCAL__DNS__]:8080`。
```
Comment (tengqm) [inline content/zh/docs/tasks/administer-cluster/nodelocaldns.md:129]:
```suggestion
  如果你使用上述的样本清单，需要把 [配置行 L70](https://github.com/kubernetes/kubernetes/blob/b2ecd1b3a3192fbbe2b9e348e095326f51dc43dd/cluster/addons/dns/nodelocaldns/nodelocaldns.yaml#L70)
  修改为 `health [__PILLAR__LOCAL__DNS__]:8080`。
```

File: content/zh/docs/tasks/administer-cluster/nodelocaldns.md [hunk 5]
```diff
@@ -103,57 +104,100 @@ This is the path followed by DNS Queries after NodeLocal DNSCache is enabled:
...
+     In this mode, node-local-dns pods listen on both the kube-dns service IP as well as `<node-local-address>`, so pods can lookup DNS records using either IP address.
+-->   
+   * 如果 kube-proxy 运行在 IPTABLES 模式：
+
+     ``` bash
+     sed -i "s/__PILLAR__LOCAL__DNS__/$localdns/g; s/__PILLAR__DNS__DOMAIN__/$domain/g; s/__PILLAR__DNS__SERVER__/$kubedns/g" nodelocaldns.yaml
+     ```
+
+     node-local-dns Pods 会设置 `__PILLAR__CLUSTER__DNS__` 和 `__PILLAR__UPSTREAM__SERVERS__`。
+     在此模式下, node-local-dns Pods 会同时侦听 kube-dns 服务的 IP 地址和 `<node-local-address>` 的地址，以便 Pods 可以使用其中任何一个 IP 地址来查询 DNS 记录。
```
Comment (tengqm) [inline content/zh/docs/tasks/administer-cluster/nodelocaldns.md:167]:
```suggestion
     在此模式下, node-local-dns Pods 会同时侦听 kube-dns 服务的 IP 地址和 `<node-local-address>` 的地址，
     以便 Pods 可以使用其中任何一个 IP 地址来查询 DNS 记录。
```

File: content/zh/docs/tasks/administer-cluster/nodelocaldns.md [hunk 6]
```diff
@@ -103,57 +104,100 @@ This is the path followed by DNS Queries after NodeLocal DNSCache is enabled:
...
 -->
-可以在任何 K8s 版本中使用上面指定的 yaml 应用该插件。
-功能支持如下所述：
+   * 如果 kube-proxy 运行在 IPVS 模式：
 
+     ``` bash
+     sed -i "s/__PILLAR__LOCAL__DNS__/$localdns/g; s/__PILLAR__DNS__DOMAIN__/$domain/g; s/__PILLAR__DNS__SERVER__//g; s/__PILLAR__CLUSTER__DNS__/$kubedns/g" nodelocaldns.yaml
+     ```
+
+     在此模式下，node-local-dns Pods 只会侦听 `<node-local-address>` 的地址。node-local-dns 接口不能绑定 kube-dns 的集群 IP 地址，因为 IPVS 负载均衡使用的接口已经占用了该地址。
```
Comment (tengqm) [inline content/zh/docs/tasks/administer-cluster/nodelocaldns.md:183]:
```suggestion
     在此模式下，node-local-dns Pods 只会侦听 `<node-local-address>` 的地址。
     node-local-dns 接口不能绑定 kube-dns 的集群 IP 地址，因为 IPVS 负载均衡
     使用的接口已经占用了该地址。
```

**Review by tengqm — COMMENTED** [review]
look good to me other than some nits.

**Comment by chenxuc** [comment]
@tengqm thanks for the review. I've updated and squashed the commit.

**Review by ydFu — APPROVED** [review]

**Comment by chenxuc** [comment]
@tengqm 
This PR and another PR 26257 haven't been merged yet although they are approved. The tide tool show message "Needs lgtm label". Do we need to add lgtm label to them? Thanks.

**Comment by tengqm** [comment]
@chenxuc Usually I'd quick approve/lgtm trivial changes which I'm 100% sure I can sign off it. That is not a good practice though, because it discourages the participation from other reviewers/approvers. For changes a little bit larger, I'd either lgtm or approve, and wait for some other team members to jump in and give it a double check. Does this makes sense to you?

**Comment by chenxuc** [comment]
> @chenxuc Usually I'd quick approve/lgtm trivial changes which I'm 100% sure I can sign off it. That is not a good practice though, because it discourages the participation from other reviewers/approvers. For changes a little bit larger, I'd either lgtm or approve, and wait for some other team members to jump in and give it a double check. Does this makes sense to you?

@tengqm Thanks for the detailed explanation. This makes 100% sense to me. I'm new to the workflow in k8s repo, so I ask more questions. Appreciate for helping me to get familiar with it.

**Comment by ydFu** [comment]
@chenxuc
Here is contributors/guide, it can help to know a lot in contributors workflow
[contributors/guide](https://github.com/kubernetes/community/blob/master/contributors/guide/owners.md)

**Comment by chenxuc** [comment]
> @chenxuc
> Here is contributors/guide, it can help to know a lot in contributors workflow
> [contributors/guide](https://github.com/kubernetes/community/blob/master/contributors/guide/owners.md)

@ydFu Thanks Ader for the tips.

**Pull request merged**